amespahdbpythonsuite/parsers.py

```diff
@@ -0,0 +1,543 @@
+#!/usr/bin/env python3
+"""
+parsers.py
+
+Parse a PAHdb XML file, with or without schema checking.
+"""
+
+import array
+import base64
+import os
+import urllib.request
+
+from lxml import etree as ElementTree
+import numpy as np
+
+
+class XMLparser:
+    """Parse a PAHdb XML file.
+
+    Optional behaviour includes validating the XML schema.
+
+    Attributes:
+        check_schema: Whether to validate the XML schema or not.
+        filename: XML file location. If not given, it is assigned to the
+            "AMESPAHDEFAULTDB" environment variable.
+
+    Note:
+        `validated` (bool) indicates if the schema is valid (True) or
+            not (False). Defaults to False until validation is performed.
+    """
+
+    def __init__(self, filename=None, check_schema=True):
+        """Inits XMLparser with schema checking on, no given filename."""
+        self.filename = filename
+        self.check_schema = check_schema
+        self.validated = False  # Assume schema has not yet been checked.
```

Comment:
This belongs in amespahdbpythonsuite.py; you should be able to validate more than once; especially if the default lxml-api is exposed.